src/react-components/room/useAvatarVolume.js

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useEffect, useState } from "react";
+import { getPlayerInfo } from "../../utils/component-utils";
+import {
+  calcLevel,
+  calcLevelStepDown,
+  calcLevelStepUp,
+  calcGainMultiplier,
+  MAX_VOLUME_LABELS
+} from "../../utils/avatar-volume-utils";
+
+const minLevel = 0;
+const maxLevel = MAX_VOLUME_LABELS - 1;
+
+export default function useAvatarVolume(sessionId) {
+  const playerInfo = getPlayerInfo(sessionId);
+  const controlsEl = playerInfo?.el.querySelector("[avatar-volume-controls]");
+  const controls = controlsEl?.components["avatar-volume-controls"];
+  const [levelStep, setLevelStep] = useState(0);
```

Comment:
The concept of "levels" here seems to have leaked from the fact that I happend to use | characters as a quick hack to display volume in media-video... It feels a bit awkward that using `useAvatarVolume` you end up reading a value in  "levels", but then writing in "gainMultiplier" units. I think everything here should just be based on gain multiplier. The levels thing should only apply at the very edge when you want to display it as a series of bars (which actually seems to be sidestepping this code anyway).

Reply:
Yep, it makes sense. I've refactored all the level calculations outside `useAvatarVolume`.